### ISSUE TITLE
Come up with a policy testing strategy

### ISSUE THREAD

**Issue by pabrahamsson** [redhat-cop/rego-policies#6]
How should a new or updated policy be tested to verify it does the right thing?
This will also be helpful for reviewing PRs and can be used for CI as well.

**Comment by garethahealy** [comment]
Already done some thinking/experiments about this. Current idea is conftest + bats.

```
@test "k8s-validation-rolebinding" {
    run conftest test _test/k8s-validation-rolebinding --output tap

    [ "$status" -eq 1 ]
    [ "${lines[1]}" = "not ok 1 - _test/k8s-validation-rolebinding/list.yml - RoleBinding/NoApiGroup: RoleBinding roleRef.apiGroup key is null, use rbac.authorization.k8s.io instead." ]
    [ "${lines[2]}" = "not ok 2 - _test/k8s-validation-rolebinding/list.yml - RoleBinding/NoKind: RoleBinding roleRef.kind key is null, use ClusterRole or Role instead." ]
    [ "${lines[3]}" = "# Successes" ]
}
```

Each test has 1 more yaml files, i.e.: _test/k8s-validation-rolebinding
```
---
apiVersion: v1
kind: List
items:
- apiVersion: rbac.authorization.k8s.io/v1
  kind: RoleBinding
  metadata:
    name: NoApiGroup
  roleRef:
- apiVersion: rbac.authorization.k8s.io/v1
  kind: RoleBinding
  metadata:
    name: NoKind
  roleRef:
    apiGroup: rbac.authorization.k8s.io
```

bats can be run from an gh-action or bash script in the repo.

**Comment by pabrahamsson** [comment]
Now that #8 has been merged I think all we need is to document the testing workflow, either in the README.md or a separate TESTING.md

**Comment by garethahealy** [comment]
@pabrahamsson ; sure, will do

**Comment by pabrahamsson** [comment]
Resolved by #10
